src/dtgtk/thumbnail.c

```diff
@@ -418,7 +421,9 @@ static gboolean _event_image_draw(GtkWidget *widget, cairo_t *cr, gpointer user_
       if(res)
       {
         // if the image is missing, we reload it again
-        g_timeout_add(250, _thumb_expose_again, widget);
+        if(!thumb->expose_again_timeout_id)
+          thumb->expose_again_timeout_id = g_timeout_add(10, _thumb_expose_again, thumb);
```

Comment:
The 10ms for the timeout seems a bit low for a thumb draw which is time consuming operation. If it was not ready then we certainly need some more time before trying again. Any explanation as to why to set it to 10 instead of 250 ?

Reply:
My understanding here is that the draw signal handler would be called repeatedly (in `expose_again`) until the thumbnail is actually drawn. The previous implementation could have added multiple timeouts for `expose_again` scheduled at the same time depending on user input (i.e. when a user triggers a draw event). Now the rule "at any time, a thumbnail can only have at most 1 expose_again scheduled" is enforced, so I think it is safe to check more frequently if the drawing is complete. Keep using 250 wouldn't cause any errors though.

Reply:
@cthbleachbit : sure I understand that, but calling more time without actually doing something is a waste of CPU time :) I would certainly restore to 250 except if there is a good reason. So please restore the value and if we have issue with that (too slow redrawing or whatever) let's re-discuss this. Works for you ?

Reply:
I admit that 10ms is probably too short to account for all I/O and stuff. However, the exact reason I changed it from 250 to 10 is that it took perceivably longer for the thumbnails to show up than before. I'll restore 250 for now, and see if there are some other confounding variables in there.